TIL.md

```diff
@@ -0,0 +1,263 @@
+# Tuesday
+
+## ember-cli-sass
+
+## Useful testing packages
+https://github.com/cibernox/ember-native-dom-helpers
+https://github.com/simplabs/ember-test-selectors
+http://ember-cli-page-object.js.org/docs/v1.8.x/
+
+## Components
+- new components were added
+  - edit which is a form because of convention and proper setting of route in route.js, 
+  ```` js
+    this.route('products', function() {
+    this.route('edit', { path: '/:product_id/edit' });,
+  ````
+    explicit search for the product was not necessary (`:product_id` is very important in this)
+- actions like `save() are declared like that, in component.js
+
+  ```` js
+  export default Component.extend({
+    actions: {
+      save() {
+        this.product.save()
+          .catch((error) => alert(error));
+      }
+    }
+  });
+  ````
+
+## Integration component test
+
+- integration tests ensure that component renders properly with given data, this cannot test transition between pages because there is no router in place 
+ 
+  ```` js
+  setupComponentTest('product-edit', {
+    integration: true
+  });
+
+  it('renders', function() {
+    this.set('product', { name: 'Phone', description: 'Black', sku: 1 });
+    this.render(hbs`{{product-edit product=product}}`);
+    expect(find('[data-test-product-form-name]').value).to.include('Phone');
+  });
+  ````
+## Acceptance test
+- integration tests ensure the application is working properly
+- without the use of page object
+  ```` js
+    ...
+    await visit('/products');
+    await click('[data-test-product-thumb-link]');
+    expect(find('[data-test-product-show-name]').textContent)
+      .to.include('Personal Computer');
+    await click('[data-test-products-link]');
+    ...
+  ````
+- with page object 
+  ```` javascript
+  // pages/product.js
+    import {
+      create,
+      visitable,
+      clickable,
+      text
+    } from 'ember-cli-page-object';
+
+    export default create({
+      visit: visitable('/products'),
+      openProduct: clickable('[data-test-product-thumb-link]'),
+      productName: text('[data-test-product-show-name]'),
+      openProducts: clickable('[data-test-products-link]')
+    });
+  ````
+
+  ```` javascript  
+  ...
+    await page.visit();
+    await page.openProduct();
+    expect(page.productName).to.include('Personal Computer');
+  ...
+  ````
+(we ensure the application is working properly, the process of transition from .. to ..)
+
+# Monday
+
+## Routing, template, components
+
+for picture, look in /TIL (ember.concepts.png)
+
+- router - is the best place to load data connected to the application path (/products - all products, /proudcts/1 - one product)
+- products router - loads data using model() hook (all products)
+- products template - we invoke component product-thumb for each element of products collection(available through model) with param product=product
+- product-thumb component - has access to product inside template.hbs and component.js (this.product)
+
+## Mirage
```

Comment:
Add link and checkbox (that you can check after reading the documentation)
[ ] [ember cli mirage](http://www.ember-cli-mirage.com/docs/v0.3.x/)